connect/resource/automation.py

```diff
@@ -43,17 +43,17 @@ def process_request(self, request):
     @function_log
     def approve(self, pk, data):
         # type: (str, dict) -> str
-        return self.api.post(path=pk + '/approve/', data=data if data else {})
+        return self.api.post(path=pk + '/approve/', data=json.dumps(data))
 
     @function_log
     def inquire(self, pk):
         # type: (str) -> str
-        return self.api.post(path=pk + '/inquire/', data={})
+        return self.api.post(path=pk + '/inquire/', data='{}')
 
     @function_log
     def fail(self, pk, reason):
         # type: (str, str) -> str
-        return self.api.post(path=pk + '/fail/', data={'reason': reason})
+        return self.api.post(path=pk + '/fail/', data=json.dumps({'reason': reason}))
```

Comment:
what you like more: 
..., data={ 'reason': reason} )
or
..., data=dict(reason=reason) )

Reply:
I tend to like more using braces for dictionaries instead of the dict function, but if you prefer the other way, I can change it, either is fine.